### ISSUE TITLE
🐛 Bug: Don't flag namespaces that only produce types (no value emit)

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [JoshuaKGoldberg/eslint-plugin-erasable-syntax-only#22]
### Bug Report Checklist

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have pulled the latest `main` branch of the repository.
- [x] I have [searched for related issues](https://github.com/JoshuaKGoldberg/eslint-plugin-erasable-syntax-only/issues?q=is%3Aissue) and found none that matched my issue.

### Expected

`--erasableSyntaxOnly` only targets namespaces that include values. The following is allowed:

```ts
namespace TypesOnly {
  export type A = 'A';
}
```

### Actual

The `erasable-syntax-only/namespaces` rule still flags the `TypesOnly` namespace. It shouldn't.

### Additional Info

💖

Noted by @jakebailey in https://bsky.app/profile/jakebailey.dev/post/3lif44jcvfs2e. Therefore:

Co-authored-by: @jakebailey

**Comment by JoshuaKGoldberg** [comment]
@all-contributors please add @jakebailey for bug.

> 🤖 Beep boop! This comment was added automatically by [all-contributors-auto-action](https://github.com/marketplace/actions/all-contributors-auto-action).
> Not all contributions can be detected from Git & GitHub alone. Please comment any missing contribution types this bot missed.
> ...and of course, thank you for contributing! 💙
